### PR TITLE
Add DNS TXT and CAA record for API-staging-cd crossfeed

### PR DESCRIPTION
**Pull request by cduhn17**
Add DNS TXT and CAA record for api.staging-cd

#  #

## 🗣 Description ##

We need to update the TXT and CAA record for the DNS to create a certificate for Let'sEncrypt. terraform-docs has been run.

Add/Update the crossfeed_api_staging_cd_TXT

## 💭 Motivation and context ##

There are no existing ACME certs

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by dav3r** [comment]
Please update this PR so that it is up-to-date with the `develop` branch and then I will review it.

**Comment by cduhn17** [comment]
> Please update this PR so that it is up-to-date with the `develop` branch and then I will review it.

The update is complete

**Comment by dav3r** [comment]
I see you are adding a new record `aws_route53_record.crossfeed_api_staging_cd_TXT`, but I don't see a corresponding line in the "Resources" section of the README.  `terraform-docs` should have given you that line in the README.

**Comment by cduhn17** [comment]
I got the update done to the README.md

File: README.md [hunk 1]
```diff
@@ -13,14 +13,14 @@ zone.  This role has a trust relationship with the users account.
 1. Run the command `terraform apply`.
 
 <!-- BEGIN_TF_DOCS -->
-## Requirements ##
+## Requirements
```
Comment (dav3r) [inline README.md]: Revert this change that `terraform-docs` makes:
```suggestion
## Requirements ##
```

In order to prevent this from happening, you need to install a specific forked branch of `terraform-docs`.  See here for details: https://github.com/cisagov/cool-dns-cyber.dhs.gov/blob/cd20047d480ccbf302e014a90418ed697b572b44/.github/workflows/build.yml#L142-L158

File: README.md [hunk 2]
```diff
@@ -13,14 +13,14 @@ zone.  This role has a trust relationship with the users account.
...
 
 | Name | Version |
 |------|---------|
 | terraform | ~> 1.0 |
 | aws | ~> 4.9 |
 
-## Providers ##
+## Providers
```
Comment (dav3r) [inline README.md]: Revert this change that terraform-docs makes:
```suggestion
## Providers ##
```

File: README.md [hunk 3]
```diff
@@ -31,13 +31,13 @@ zone.  This role has a trust relationship with the users account.
 | aws.route53resourcechange | ~> 4.9 |
 | terraform | n/a |
 
-## Modules ##
+## Modules
```
Comment (dav3r) [inline README.md]: Revert this change that terraform-docs makes:
```suggestion
## Modules ##
```

File: README.md [hunk 4]
```diff
@@ -31,13 +31,13 @@ zone.  This role has a trust relationship with the users account.
...
 
 | Name | Source | Version |
 |------|--------|---------|
 | read\_terraform\_state | github.com/cisagov/terraform-state-read-role-tf-module | n/a |
 
-## Resources ##
+## Resources
```
Comment (dav3r) [inline README.md]: Revert this change that terraform-docs makes:
```suggestion
## Resources ##
```

File: README.md [hunk 5]
```diff
@@ -166,7 +167,7 @@ zone.  This role has a trust relationship with the users account.
 | [terraform_remote_state.pca_staging](https://registry.terraform.io/providers/hashicorp/terraform/latest/docs/data-sources/remote_state) | data source |
 | [terraform_remote_state.terraform](https://registry.terraform.io/providers/hashicorp/terraform/latest/docs/data-sources/remote_state) | data source |
 
-## Inputs ##
+## Inputs
```
Comment (dav3r) [inline README.md]: Revert this change that terraform-docs makes:
```suggestion
## Inputs ##
```

File: README.md [hunk 6]
```diff
@@ -186,7 +187,7 @@ zone.  This role has a trust relationship with the users account.
 | sessendemail\_role\_name | The name to assign the IAM role (as well as the corresponding policy) that allows sufficient permissions to send email via AWS SES. | `string` | `"SesSendEmail-cyber.dhs.gov"` | no |
 | tags | Tags to apply to all AWS resources created. | `map(string)` | `{}` | no |
 
-## Outputs ##
+## Outputs
```
Comment (dav3r) [inline README.md]: Revert this change that terraform-docs makes:
```suggestion
## Outputs ##
```

**Review by dav3r — CHANGES_REQUESTED** [review]
In order to prevent `terraform-docs` from removing the closing `##` in the README headers, you need to install a specific forked branch of `terraform-docs`.  See here for details: https://github.com/cisagov/cool-dns-cyber.dhs.gov/blob/cd20047d480ccbf302e014a90418ed697b572b44/.github/workflows/build.yml#L142-L158

File: README.md (hunk 6 above)
Comment (cduhn17) [reply README.md]: I just pushed to fix. The `##` are getting removed when I i run `terraform fmt`

File: README.md (hunk 1 above)
Comment (dav3r) [reply README.md]: Fixed in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/103/commits/5de085cd1da6a42095891d88144f4bf0b14543a4.

File: README.md (hunk 3 above)
Comment (dav3r) [reply README.md]: Fixed in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/103/commits/5de085cd1da6a42095891d88144f4bf0b14543a4.

File: README.md (hunk 4 above)
Comment (dav3r) [reply README.md]: Fixed in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/103/commits/5de085cd1da6a42095891d88144f4bf0b14543a4.

File: README.md (hunk 5 above)
Comment (dav3r) [reply README.md]: Fixed in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/103/commits/5de085cd1da6a42095891d88144f4bf0b14543a4.

File: README.md (hunk 6 above)
Comment (dav3r) [reply README.md]: Fixed in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/103/commits/5de085cd1da6a42095891d88144f4bf0b14543a4.

File: README.md (hunk 2 above)
Comment (dav3r) [reply README.md]: Fixed in https://github.com/cisagov/cool-dns-cyber.dhs.gov/pull/103/commits/5de085cd1da6a42095891d88144f4bf0b14543a4.

**Review by dav3r — APPROVED** [review]
Looks good now.  I went ahead and applied this change, so your TXT record should now be visible in DNS.

**Comment by cduhn17** [comment]
@dav3r , 
I had a failure at LE. I had to add a CAA to allow digicert CA.

**Review by dav3r — APPROVED** [review]
👍  I re-applied to update the TXT record and create that CAA record.  Please let me know if you need anything else for this PR.

**Comment by cduhn17** [comment]
That worked thanks for all your assistance!

**Review by jsf9k — APPROVED** [review]
